File: backend/src/app/schemas/goal.py
from uuid import UUID

from app.models.nutrition import Goal
from pydantic import BaseModel


class CreateGoalRequest(BaseModel):
    goal_type: Goal


class GoalResponse(BaseModel):
    id: UUID
    goal_type: Goal
    target_kcal: float
    protein_g: float
    fat_g: float
    carbs_g: float

File: frontend/src/types/goal.ts
export type GoalType = 'diet' | 'strength' | 'bouldering'

export interface CreateGoalRequest {
  goal_type: GoalType
}

export interface GoalResponse {
  id: string
  goal_type: GoalType
  target_kcal: number
  protein_g: number
  fat_g: number
  carbs_g: number
}
